W-10590530-mel-parsing-issue-v3.9-cv (#2160)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/mule-expression-language-tips.adoc

+
Note also that, if given the expression `#[flowVars.abc.toString()]` and the value of '`abc`' is `null`, Mule throws a `NullPointerException`.

> === Evaluation Gotchas

Reviewer: ```suggestion
=== Evaluation Troubleshooting
```

> There is a known issue that causes some incorrect expressions to work during an evaluation without causing any compiler errors. The issue occurs when an unexpected character is affixed at the end of an escaped property in an expression that is part of an evaluation, for example:

Reviewer: ```suggestion
A known issue causes certain incorrect expressions to be accepted during an evaluation without causing any compiler errors. The issue occurs when an unexpected character is affixed at the end of an escaped property in an expression that is part of an evaluation, for example:
```
work > be accepted? succeed? used?

```
<when expression="#[message.inboundProperties.'http.request.path'] == '/some_path']">
```

> The previous expression is incorrect, because a `]` character is incorrectly affixed to the `'http.request.path'` property. However, Mule ignores this extra character and is able to compile the expression without errors.

Reviewer: ```suggestion
Although the `]` character is incorrectly affixed to the `'http.request.path'` property in this example, Mule ignores this extra character and is able to compile the expression without errors.
```

> Keep in mind this parsing issue while troubleshooting incorrect expressions or if you experience unexpected behaviors as a result of evaluating an expression. This behavior is present in all Mule 3 versions, in any component that evaluates a MEL expression.

Reviewer: Does Mule's ability to ignore extra characters ever cause any problems? From what I read here, although it might be considered a bug that Mule can do this, no harm is done...so why are we documenting this? Seems like behavior that does not impact users?

== See Also

> * For the complete MEL reference, including lists of operators, imported Java classes, context objects, etc. see xref:mule-expression-language-reference.adoc[Mule Expression Language Reference].

Reviewer: ```suggestion
* For the complete MEL reference, including lists of operators, imported Java classes, context objects, and so on, see xref:mule-expression-language-reference.adoc[Mule Expression Language Reference].
```

> * Quick tips for MEL xref:mule-expression-language-tips.adoc[Mule Expression Language Tips]

Reviewer: * For a quick reference to MEL, see xref:mule-expression-language-tips.adoc[Mule Expression Language Tips].
